tests/test_engines.py

```diff
@@ -66,3 +68,21 @@ def test_tor_get_source_valid_url(self):
             response = self.engine.get_page_source(url)
             assert isinstance(response, Webpage)
             assert isinstance(response.source, str)
+
+    def test_call_limiter_before_limit(self):
```

Comment:
Unless we can come up with a reasonable way to mock out the controller connection object, this is about as much as I can test.